### PR TITLE
[WIP] Add restore button to deleted events

### LINKED ISSUE #1483: Add restore button to deleted events
**Describe the bug**

Events once deleted should have a restore button.

**To Reproduce**
1. Go to admin/events
2. Click on delete
3. See error

**Expected behavior**

Once deleted events should have a restore button which when clicked should restore that event.

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser: Chromium

### PR DESCRIPTION
**Pull request by ritikamotwani**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Adds a restore button to deleted events. 

#### Changes proposed in this pull request:

- In admin/events all the events should have a restore button after they are deleted once.

Fixes #1483

### CODE REVIEW

**Comment by ritikamotwani** [comment]
@niranjan94 @CosmicCoder96 @pradeepgangwar @srv-twry I am facing some problem in this issue. Although I have added a filter and done get_trashed=true. and in the controller's get request I get 
image
But for the patch request I get 
image

**Comment by srv-twry** [comment]
@ritikamotwani I can't see the `?get_trashed=true` in the URL given in the screenshot.

**Comment by ritikamotwani** [comment]
@srv-twry that's what I am saying I have written it in the controller. The first screenshot I get it when .set(deleted at,null) is called. But when event.save is called that is a patch request the filter and the value of get_trashed disappear, the second screenshot.

**Comment by srv-twry** [comment]
oh :/
I though you're pointing at a bug in the server.

File: app/controllers/admin/events/list.js [hunk 1]
```diff
@@ -65,6 +65,18 @@ export default Controller.extend({
...
       this.transitionToRoute('events.view.edit.basic-details', id);
     },
+    restoreEvent(id) {
+      this.store.query('event', {
+        filter: [{
+          name : 'identifier',
+          op   : 'eq',
+          val  : id
+        }],
+        get_trashed: true }).then(function(event) {
```
Comment (CosmicCoder96) [inline app/controllers/admin/events/list.js]: have you tried passing `get_trashed` as a filter?
like,
```js
{ 
name : 'get_trashed',
op: 'eq',
val: true
}
```

Comment (ritikamotwani) [reply app/controllers/admin/events/list.js]: @CosmicCoder96 I tried this but it is giving an error.
image

Comment (Pastisa) [reply app/controllers/admin/events/list.js]: https://user-images.githubusercontent.com/22127980/44157039-11140800-a0cf-11e8-97e6-c8e86fc77075.png

**Pull request closed without merging**